### PR TITLE
feat: Add filter processor

### PR DESCRIPTION
**Pull request by biblicalph**
Implement support for filter processor.

Ref: [LOG-18377](https://mezmo.atlassian.net/browse/LOG-18377)

[LOG-18377]: https://mezmo.atlassian.net/browse/LOG-18377?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: examples/processors/filter.tf [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  }
+}
+
+resource "mezmo_logs_destination" "destination1" {
+  pipeline_id   = mezmo_pipeline.pipeline1.id
+  title         = "My destination"
+  description   = "Send logs to Mezmo Log Analysis"
+  inputs        = [mezmo_filter_processor.processor1.id]
+  ingestion_key = var.my_ingestion_key
+}
```
Comment (darinspivey) [inline examples/processors/filter.tf:52]: Needs a blank line at the end of the file.

File: examples/processors/filter.tf [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  conditionals = {
+    expressions = [
+      {
+        field        = ".status"
+        operator     = "greater_or_equal"
+        value_number = 300
+      },
+      {
+        field        = ".level"
+        operator     = "greater_or_equal"
```
Comment (darinspivey) [inline examples/processors/filter.tf:38]: I don't think >= is the right operator for a string

File: examples/processors/filter.tf [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    logical_operation = "OR"
+  }
+}
+
+resource "mezmo_logs_destination" "destination1" {
+  pipeline_id   = mezmo_pipeline.pipeline1.id
+  title         = "My destination"
+  description   = "Send logs to Mezmo Log Analysis"
+  inputs        = [mezmo_filter_processor.processor1.id]
+  ingestion_key = var.my_ingestion_key
```
Comment (darinspivey) [inline examples/processors/filter.tf:51]: I don't think we need this.  It doesn't add value for understanding the new processor, and it's another place we'd have to update if the mezmo_logs destination ever changes.

File: internal/provider/models/processors/test/filter_test.go [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+							expressions = [
+								{
+									field = ".status"
+									operator = "equal"
+									value_number = 200
+								}
+							]
+						}
+					}`,
+				ExpectError: regexp.MustCompile("Attribute action value must be one of: \\[\"allow\" \"drop\"\\], got: \"unknown\""),
```
Comment (darinspivey) [inline internal/provider/models/processors/test/filter_test.go:77]: tip: Use backticks for these strings and you only have to escape the bracket once:

https://github.com/mezmo/terraform-provider-mezmo/blob/e66182e255cabed1ffbf4b03642e04d9ad6903af/internal/provider/models/processors/test/metrics_tag_cardinality_limit_test.go#L108

File: internal/provider/models/processors/test/filter_test.go [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+											operator = "equal"
+											value_number = 200
+										}
+									]
+									logical_operation = "AND"
+								}
+							]
+						}
+					}`,
+				ExpectError: regexp.MustCompile("(?s)Attribute \"conditional.expressions_group\" cannot be specified when.*\"conditional.expressions\" is specified"),
```
Comment (darinspivey) [inline internal/provider/models/processors/test/filter_test.go:116]: Ditto.  backticks allow double quotes to be unescaped.

File: internal/provider/models/processors/test/filter_test.go [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+						"conditional.expressions_group.1.expressions_group.0.expressions.0.field":        ".level",
+						"conditional.expressions_group.1.expressions_group.0.expressions.0.operator":     "greater_or_equal",
+						"conditional.expressions_group.1.expressions_group.0.expressions.0.value_number": "300",
+						"conditional.expressions_group.1.expressions_group.0.expressions.1.field":        ".tag",
+						"conditional.expressions_group.1.expressions_group.0.expressions.1.operator":     "contains",
+						"conditional.expressions_group.1.expressions_group.0.expressions.1.value_string": "error",
+					}),
+				),
+			},
+			// Connect outputs to destinations
```
Comment (darinspivey) [inline internal/provider/models/processors/test/filter_test.go:245]: Why do we have to test this?  Connecting nodes is normal vector stuff, not having to do with this processor.  Am I missing something?

File: pkg/resources/convertible.go [hunk 7]
```diff
@@ -29,14 +29,17 @@ func ConvertibleResources() ([]ConvertibleResourceDef, error) {
...
 		tfRes := tfResFn()
 		cRes, ok := tfRes.(ConvertibleResourceDef)
+		keys = append(keys, cRes.NodeType())
 		if !ok {
 			err := fmt.Errorf("terraform resource %T does not implement ConvertibleResourceDef", tfRes)
 			return nil, err
 		}
 		convertibleRes[i] = cRes
 	}
+	fmt.Println(keys)
```
Comment (darinspivey) [inline pkg/resources/convertible.go:43]: Intentional?

File: internal/provider/models/processors/test/filter_test.go (hunk 6 above)
Comment (biblicalph) [reply internal/provider/models/processors/test/filter_test.go:245]: Twas mostly for my sanity.  I wondered for a moment if filter had multiple outputs

File: pkg/resources/convertible.go (hunk 7 above)
Comment (biblicalph) [reply pkg/resources/convertible.go]: Nope. Dumped keys out for Terry and got bundled here.

**Review by darinspivey — APPROVED** [review]
nice!

**Pull request merged**